pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -262,8 +296,146 @@ func (c *cyclonedxParser) GetPredicates(ctx context.Context) *assembler.IngestPr
 	return preds
 }
 
-func (s *cyclonedxParser) getPackageElement(elementID string) []*model.PkgInputSpec {
-	if packNode, ok := s.packagePackages[string(elementID)]; ok {
+func (c *cyclonedxParser) getVulnerabilities(ctx context.Context, cdxBom *cdx.BOM) *vulnData {
+	if cdxBom.Vulnerabilities == nil {
+		return nil
+	}
+
+	var vInfo vulnData
+	var status model.VexStatus
+	var justification model.VexJustification
+	var publishedTime time.Time
+	layout := "2006-01-02T15:04:05.000Z"
+	for _, vulnerability := range *c.cdxBom.Vulnerabilities {
+		vuln, err := asmhelpers.CreateVulnInput(vulnerability.ID)
+		if err != nil {
+			return nil
+		}
+
+		if vexStatus, ok := vexStatusMap[vulnerability.Analysis.State]; ok {
+			status = vexStatus
+		}
+
+		if vexJustification, ok := justificationsMap[vulnerability.Analysis.Justification]; ok {
+			justification = vexJustification
+		}
+
+		time, err := time.Parse(layout, vulnerability.Published)
+		if err == nil {
+			publishedTime = time
```

Comment:
this would result in a default value for time? Seems like a bad approach.

Reply:
This is just for converting string time to time.Time using a predefined layout, any alternatives in mind?

Reply:
lets move all these to the [parse function[(https://github.com/guacsec/guac/blob/76e3ff907c0f0c81d4c76017a9ea4b606f307baa/pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go#L74C27-L74C320) so we can return an error and just return an error if it does not evaluate.

Reply:
similar to what @jeffmendoza said